cudamapper/src/minimizer.cu

```diff
@@ -836,32 +839,34 @@ __global__ void find_back_end_minimizers(const std::uint64_t minimizer_size,
 /// Many windows share the same minimizer and each minimizer is written only once, meaning many elements do not contain minimizers.
 /// This function creates new arrays where such elements do not exist.
-/// Note that in the input arrays all minimizers of one read are written consecutively, i.e. [read 0 minimizers], [read 0 junk], [read 1 minimizers], [read 1 junk], [read 2 minimizers]...
+/// Note that in the input arrays all minimizers of one read are written consecutively, i.e. [read_0 minimizers], [read_0 junk], [read_1 minimizers], [read_1 junk], [read_2 minimizers]...
 ///
 /// \param window_minimizers_representation array of representations of minimizers, grouped by reads
 /// \param window_minimizers_position_in_read array of positions in read of minimizers, grouped by reads
 /// \param window_minimizers_direction array of directions of minimizers, grouped by reads (0 - forward, 1 - reverse)
 /// \param read_id_to_windows_section index of first element dedicated to that read in input arrays and the number of dedicated elements
 /// \param representations_compressed array of representations of minimizers, grouped by reads, without invalid elements between the reads
 /// \param rest_compressed array of read_ids, positions_in_read and directions of reads, grouped by reads, without invalid elements between the reads
-/// \param read_id_to_compressed_minimizers index of first element dedicated to that read in input arrays and the number of dedicated elements
+/// \param read_id_to_compressed_minimizers index of the first minimizer of the next read (array comes from an inclusive scan, hence all indices are shifted by one)
+/// \param read_id_of_first_read
 __global__ void compress_minimizers(const representation_t* const window_minimizers_representation,
                                     const position_in_read_t* const window_minimizers_position_in_read,
                                     const char* const window_minimizers_direction,
                                     const ArrayBlock* const read_id_to_windows_section,
                                     representation_t* const representations_compressed,
                                     Minimizer::ReadidPositionDirection* const rest_compressed,
-                                    const ArrayBlock* const read_id_to_compressed_minimizers,
-                                    std::uint32_t offset)
+                                    const std::int64_t* const read_id_to_compressed_minimizers,
+                                    std::uint32_t read_id_of_first_read)
 {
-    const auto& first_input_minimizer  = read_id_to_windows_section[blockIdx.x].first_element_;
-    const auto& first_output_minimizer = read_id_to_compressed_minimizers[blockIdx.x].first_element_;
-    const auto& number_of_minimizers   = read_id_to_compressed_minimizers[blockIdx.x].block_size_;
+    const auto& first_input_minimizer = read_id_to_windows_section[blockIdx.x].first_element_;
```

Comment:
Optional: while you are at it, could you remove the reference?

Reply:
done